Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt

```diff
@@ -106,13 +113,15 @@ class ConfidenceFeatureProvider private constructor(
         oldContext: EvaluationContext?,
         newContext: EvaluationContext
     ) {
-        if (newContext != oldContext) {
-            // on the new context we want to fetch new values and update
-            // the storage & cache right away which is why we pass `InitialisationStrategy.FetchAndActivate`
-            internalInitialize(
-                newContext,
-                InitialisationStrategy.FetchAndActivate
-            )
+        val context = newContext.toConfidenceContext()
+        for (entry in context.map) {
+            confidence.putContext(entry.key, entry.value)
```

Comment:
calling this in a for loop will emit multiple times on the `contextChanges` Flow. I don't think we want that.
Better to collect all of these changes in one emit.

Reply:
done in the last commit. made the update with removed keys transactional.